### DISCUSSION TITLE
difference in ordinal latent output

### DISCUSSION THREAD

**Discussion by eborgman** [lzachmann/models-for-missing-data#20] [General]
Hey @lzachmann. I've come across something curious and would like your thoughts.

I'm getting different stratum-level results when using my old 3-management-zone covariates compared to the newer 5-management-zone covariate file.  I'm trying to think through what this means and can't quite wrap my head around it.

Notice the offset in the 1st and 4th strata means; they're considerably lower with the new , 5-zone covariates. I compared the two files and don't see any changes besides the mgmt zones. and both models pull from the same source data. What does it mean that this 5-zone covariate is depressing certain site and stratum-level means?

model: assets\uplands-output\ROMN\GRKO\soil-stability\SubSampLoss\ord-lat-norm_lin_none_fixed-stratum\botanist-ja_mgmtzone-deflections

old covariate data - 3 zones
![image](https://user-images.githubusercontent.com/25830271/144676770-1739ba4b-6613-494a-83a2-36ad969e683a.png)

new covariate data - 5 zones
![image](https://user-images.githubusercontent.com/25830271/144676859-788a253c-cf3b-4ea4-bbb1-712aff42a523.png)

**Comment by lzachmann** [comment]
Ah this is very interesting. The apparent offset in the 1st and 4th strata means is a sensible place to focus. The differences are fairly dramatic, and we should be able to find an explanation for the difference(s) we see in the zone coefficients. Although we probably ought to focus on one or two strata to start, note that the new model (utilizing five zones) is not universally depressing strata means. The means in stratum 3 and stratum 5 appear to go up, for instance, with the new five-zone data.

I would place at least a small bet on both models passing posterior predictive checks at about the same level. (I'm assuming they are both converging, at any rate.) Is it true that both have the Bayesian p-values in the same general neighborhood? 🤞 

If so, we can turn to the more interesting stuff. Here's what I think may be happening, and I think we can confirm this with some basic plotting / mapping, perhaps. (Please forgive the presumptuous review of material in this first comment -- I need to remind myself how this works first and foremost).
- First, the `hat.strat` type summaries you're showing do not include the effects of any other covariates, by design. It's the mean (in this case the mean sub-surface soil stability) over time at the mean of all other covariates. If the effects of zone were modeled as 'deflections', then it's the mean sub-surface soil stability over time across _all_ management zones, whether or not a site of a certain zone even appears in a given stratum. The reason I mention this will be more clear shortly. I hope...
- Just as the effect of some covariates, say x1, can change when including a new covariate, x2 (presumably because x2 is now soaking up some of the variability x1 previously accounted for), the intercept can (and will) also change in response to the inclusion of new / different covariates. 
- I mentioned the posterior predictive checks because my guess is that -- whether we use 3 or 5 zones, we still get reasonable predictions, in which the `y.rep`s look a lot like the data we observed. But in any given stratum, the model's 'strategy' for arriving at reasonable predictions needs to change. In the case of the first stratum, using the 3-zone old covariate data, the strategy was to use a higher b0 term with perhaps less interesting zone coefficients for any zones that appeared in that stratum. In the case of the new covariate data, it's a smaller b0 coupled with what may be strong (in this case more strongly positive) zone effects for the sites/zones appearing in that stratum? To do a proper job we would need to determine the number of sites of each zone in a given stratum to appropriately weight the overall mean, which will include the intercept _and_ all zone related terms. 

Let me know what you think, and if you might be willing to help me do a little digging to unpack this! I'm sure the commentary above raises more questions than it resolves, but figured it may help us gain a little traction to begin with.

**Reply by eborgman** [reply]
Yes ppc are good for both models. Bayesian p and sd between 0.50-0.60 and yreps look like the observed data.

Your instincts seem on point here. Here are some additional figures that do seem to support your guess - higher Beta coefficients with the newer covariate data for some of these strata. (see below)

I'm interested in pursuing and understanding this because with the old covariate data, the model for soil stability came out as "good," but with the new covariate data, it would fall into the "bad" category. Park managers don't like to see aspects of their park in bad condition (especially since I've been saying it's good condition up until now!), so I want to be sure this model with the new covariates is an accurate reflection of condition.

Are you around this week to chat? I'm pretty open except from 1-2 today, except 11:30-12:30 tomorrow, and I'm free up until 2 on Friday.

_Some additional figures/tables_
5-zone (irrigated east, irrigated west, riparian, upland east, upland west)
![image](https://user-images.githubusercontent.com/25830271/145259855-1286c9c1-b206-417d-8494-e04fc232ac5e.png)

Beta[1] is botanist and the rest are for each mgmt zone
![image](https://user-images.githubusercontent.com/25830271/145260686-662c2439-f6c8-43c4-8e64-24bf0e255e69.png)

3-zone (irrigated, riparian, upland)
![image](https://user-images.githubusercontent.com/25830271/145260093-2bc11120-c4b6-46b7-bee4-f033223c3ce7.png)

![image](https://user-images.githubusercontent.com/25830271/145260766-e8f810d8-b292-44de-97a4-be4bbf715be6.png)

**Comment by lzachmann** [comment]
Hey Erin, this is definitely a case where we'd want to use the `pred.strat` object instead, because they will use the actual value of zone for each site (sampled or not) in each strata when conditioning for the prediction of the stratum level mean. It was also use the actual time-varying values of any continuous covariates, but maybe that's okay? In some sense the `hat.strat` result is uniquely non-sensible in your case. The reasons for this are implicit in some of the reasoning in my last post. We can try to be more explicit about why this is the case when we chat.

I may have messed up the scaling / rescaling of `X.pred` so I need to make sure that looks okay first, but let's discuss when you have time. I'll shoot you a text sometime (after 2) this afternoon when I have a window.
